lib/commands/source.js

```diff
@@ -13,13 +13,15 @@ const commands = {
       return await this.executeAtom('execute_script', [script, []]);
     }
 
-    if ((await this.settings.getSettings()).useJSONSource) {
-      const srcTree = await this.mobileGetSource('json');
+    const settings = await this.settings.getSettings();
+    const excludedAttributes = settings.pageSourceExcludedAttributes;
+    const hasExcludedAttributes = _.isString(excludedAttributes) && !_.isEmpty(excludedAttributes);
+
+    if (settings.useJSONSource) {
+      const srcTree = await this.mobileGetSource('json', hasExcludedAttributes ? excludedAttributes : undefined);
       return getSourceXml(getTreeForXML(srcTree));
     }
 
-    const excludedAttributes = (await this.settings.getSettings()).pageSourceExcludedAttributes;
-    const hasExcludedAttributes = _.isString(excludedAttributes) && !_.isEmpty(excludedAttributes);
     return hasExcludedAttributes
```

Comment:
this condition could also be simplified same way the above one is called: 
```
await this.mobileGetSource('xml', hasExcludedAttributes ? excludedAttributes : undefined)
```

Reply:
done